### PR TITLE
Build Load test tool for addon manager

### PR DESCRIPTION
**Pull request by ccfishk**
The tool can be run as a daemon on linux platform, generated a summary.txt file under the current directory.

go run test-load/dolphin/main.go tests --numberOfGoRoutines 10 --managerPID ### --wfcontrollerPID ###

it will generate 10 * 200 workflow (sample, eventroute.yaml) on the cluster; 
collecting add-on-manager and workflow-controller %cpu,%mem every 2 minutes
flushing the data into summary.txt
Note: some error cases need be covered

### CODE REVIEW

**Comment by kevdowney** [comment]
Please add a `make load-test` target to execute this.

**Review by tekenstam — APPROVED** [review]

File: test-load/pkg/cmd/dolphin/tests/load.go [hunk 1]
```diff
@@ -1,74 +1,85 @@
-/*
```
Comment (figo) [inline test-load/pkg/cmd/dolphin/tests/load.go:1]: I understand this already merged, but hope we can start the discussion here then address it in the following PR.

Why the copyright been removed here? also most of other source files in the PR don't have copyright file.

File: test-load/pkg/exec/helpers.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		if rErr, ok := err.(*RunError); ok {
+			runError = rErr
+		} else {
+			break
+		}
+	}
+	return runError
+}
+
+func CombinedOutputLines(cmd Cmd) (lines []string, err error) {
```
Comment (figo) [inline test-load/pkg/exec/helpers.go:34]: The golang library https://pkg.go.dev/os/exec#Cmd.CombinedOutput looks like does exactly like this, why we need to build a library?

File: test-load/pkg/internal/cli/logger.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
+package cli
```
Comment (figo) [inline test-load/pkg/internal/cli/logger.go:1]: why build a new logger?

File: test-load/pkg/internal/sets/string.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
+package sets
```
Comment (figo) [inline test-load/pkg/internal/sets/string.go:1]: why build the set? https://pkg.go.dev/github.com/golang-collections/collections/set#pkg-types

**Review by figo — COMMENTED** [review]
I understand this has been merged, but hope we can address the questions in the following PR.  cc @kevdowney @tekenstam

**Pull request merged**